src/core/coap/coap_secure.cpp

```diff
@@ -178,12 +178,12 @@ Error CoapSecure::Send(ot::Message &aMessage, const Ip6::MessageInfo &aMessageIn
     return kErrorNone;
 }
 
-void CoapSecure::HandleDtlsConnected(void *aContext, bool aConnected)
+void CoapSecure::HandleDtlsConnected(void *aContext, otCoapSecureConnectEvent aEvent)
```

Comment:
I suggest defining a `typedef` for this `enum` within the `SecureTransport` and/or `CoapSecure` class to avoid using the C-style `otCoapSecureConnectEvent` in C++ core modules.

Something like this maybe (we can define constants for the `enum` values)

```c++
class SecureTransport : public InstanceLocator
{
public:
    typedef otCoapSecureConnectEvent ConnectEvent; ///< A connect event.

    static constexpr ConnectEvent kConnected               = OT_COAP_SECURE_CONNECTED;
    static constexpr ConnectEvent kDisconnectedPeerClosed  = OT_COAP_SECURE_DISCONNECTED_PEER_CLOSED;
    static constexpr ConnectEvent kDisconnectedLocalClosed = OT_COAP_SECURE_DISCONNECTED_LOCAL_CLOSED;
    static constexpr ConnectEvent kDisconnectedMaxAttempts = OT_COAP_SECURE_DISCONNECTED_MAX_ATTEMPTS;
    static constexpr ConnectEvent kDisconnectedError       = OT_COAP_SECURE_DISCONNECTED_ERROR;

    /**
     * Function pointer which is called reporting a connection event (when connection established or disconnected)
     *
     */
    typedef otHandleCoapSecureClientConnect ConnectedHandler;
```

An alternative can be to define a new mapped  `enum` in core modules and using `MapEnum` (for example like `DefineMapEnum(otDeviceRole, Mle::DeviceRole)`. However, I am not sure if this will work since the `enum` type is used as a callback parameter.

Reply:
Thanks for the suggestion, done